lib/altsvc.c

```diff
@@ -523,7 +525,7 @@ CURLcode Curl_altsvc_parse(struct Curl_easy *data,
           if(!*p || *p == ',')
```

Comment:
The condition a few lines up looks a bit odd, if `*p` is a blank, then it is definitely not equal to `;` or `,`, so that last part of the condition could be dropped. Instead of the proposed change, what about:
``` diff
-          while(*p && ISBLANK(*p) && *p != ';' && *p != ',')
+          while(*p && ISBLANK(*p))
             p++;
-          if(!*p || *p == ',')
+          if(!*p || *p == ',' || ISNEWLINE(*p))
             break;
           p++; /* pass the semicolon */
```

Malformed headers such as `alt-svc: h2=":443"Xma=1,` would previously also be accepted. So alternatively, the above `if(!*p || *p == ',' || ISNEWLINE(*p))` could also be simplified to `if(*p != ';')`.

There are still some invalid cases that are accepted due to the check for `!ISBLANK(*p)` as opposed to checking for a "token". Maybe the parser could be made more stricter and apply these grammar:
https://tools.ietf.org/html/rfc7838#section-3
https://tools.ietf.org/html/rfc7230#section-3.2.6
Right now, any valid partial result is returned. Should it reject everything instead on a partial parsing failure?

Reply:
Oh yes, my fix was only about making sure this particular case was fixed. It did not strengthen it any other ways.

I think we should work on improving the parser by making sure we add more test cases for the cases you mention above - and more. It just feels like a separate task...

Reply:
Oh, @jay did suggest a further simplification earlier on in #5445 for the first condition, just `while(ISBLANK(*p))`.